kubeapps-dashboard.jsonnet

```diff
@@ -130,6 +138,7 @@ local serviceDeployFromValues(parentName, componentName, values) = {
                   config: {mountPath: "/monocular/config"},
                 },
               },
+              tiller: (import "tiller-deployment.jsonnet").spec.template.spec.containers[0],
```

Comment:
I see this container includes `TILLER_NAMESPACE=kube-system` env var (it doesn't use downward API).  I don't know what tiller itself uses that value for - but I guess we need to also change it to `kubeapps` as part of this move (either overriding the literal value in jsonnet or change it to use the downward API).

Reply:
I _think_ this is where Tiller stores the ConfigMaps it uses to record releases. I was thinking about just leaving this in kube-system, since it acts on the whole cluster anyway, but I guess keeping everything related to the Tiller install in the same NS makes the most sense. Will change.

Reply:
:+1: If you've tested the result, and it works, then I don't really care how the sausage was made ;)

Reply:
Now that I think about it, the answer for this depends on how compatible we want to be with the Helm CLI.

1. We want to be completely compatible with the Helm CLI
  - Make the Dashboard API a sidecar of tiller-deploy
  - Keep TILLER_NAMESPACE as "kube-system"

2. We want to run a secured Tiller, but support a user install of Tiller the normal way using `helm init`
  - Keep TILLER_NAMESPACE as "kube-system" - this will ensure release history is shared with the Helm CLI's Tiller and ours.

3. We want to completely isolate Tiller and manage Helm releases from the UI and CLI separately
  - Change TILLER_NAMESPACE to "kubeapps"
  - This will allow another Tiller to run in "kube-system", but they will each hold their own independent release history

cc @arapulido

Reply:
I have asked @sebgoa to have a look and comment here

Reply:
I vote for #3 we deploy "our" helm in `kubeapps` namespace , break the helm cli, hence force the use of "our" tiller via the UI. This should allow us to allow helm-crd CLI access as well. 

second reason is that some folks may already have helm running in the cluster and we should not mess with that (it is not kubeapps responsibility).

I let @anguslees tell us if this is reasonable and then you should merge tomorrow morning.

Reply:
sgtm, I just thought this deserved a broader discussion than a quiet PR by adnan and a green button push by me.

Reply:
but this manifest does not use helm-crd yet ?

Reply:
sgtm, will make the change to option 3